pkg/apis/application/v1alpha1/types.go

```diff
@@ -205,6 +205,11 @@ func (s ApplicationSources) Equals(other ApplicationSources) bool {
 	return true
 }
 
+// IsZero returns true if the application source is considered empty
+func (a ApplicationSources) IsZero() bool {
+	return len(a) == 0
+}
```

Comment:
Considering we need to check for Sources to be not nil before we call `IsZero` anytime, I think we can better utilize this function as below:
```suggestion
func (a ApplicationSources) IsNullOrEmpty() bool {
	return (a != nil && len(a) == 0) || a == nil
}
```

WDYT?